frontend/src/metabase-lib/v1/metadata/utils/models.ts

```diff
@@ -3,14 +3,14 @@ import type Question from "metabase-lib/v1/Question";
 import type Database from "metabase-lib/v1/metadata/Database";
 import { getQuestionVirtualTableId } from "metabase-lib/v1/metadata/utils/saved-questions";
 import type NativeQuery from "metabase-lib/v1/queries/NativeQuery";
-import { isSameField } from "metabase-lib/v1/queries/utils/field-ref";
+import { findColumnIndexesForColumnSettings } from "metabase-lib/v1/queries/utils/dataset";
```

Comment:
> Make sure you can change metadata for duplicated columns separately

It does not seem to work. It's broken in `master` as well but has slightly different symptoms.
Are you sure this PR can fix it?

https://github.com/metabase/metabase/assets/6830683/f1420a5a-9356-49ad-a151-cf9d96ca17b1

Reply:
@kamilmielnik you're correct - this PR is blocked until `column.name` is fixed and actually unique. Discussed here https://metaboat.slack.com/archives/C04CYTEL9N2/p1720016104124849

Reply:
Fixed now

Reply:
Not fixed. I reverted this change as it becomes too much for one PR